source/DSCResources/DSC_CMSoftwareUpdatePoint/DSC_CMSoftwareUpdatePoint.psm1

```diff
@@ -0,0 +1,458 @@
+
+    .PARAMETER UseProxyForAutoDeploymentRule
+        Indicates whether an auto deployment rule can use a proxy.
+
+    .PARAMETER WSUSAccessAccount
+        Specifies an account used to connect to the WSUS server. When not used, specify the AnonymousWSUSAccess parameter.
+
+        If specifying an account the account must already exist in
+        Configuration Manager.  This can be achieved by:
+
+        $secure = ConvertTo-SecureString -String "Password" -AsPlainText -Force
+        New-CMAccount -Name 'contoso\test1' -Password $secure -SiteCode '<siteCode>'
+
+    .PARAMETER WsusIisPort
+        Specifies a port to use for unsecured access to the WSUS server.
+
+    .PARAMETER WsusIisSslPort
+        Specifies a port to use for secured access to the WSUS server.
+
+    .PARAMETER WsusSsl
+        Specifies whether the software update point uses SSL to connect to the WSUS server.
+
+    .PARAMETER Ensure
+        Specifies whether the software update point is present or absent.
+
+    .Notes
+        This must be run on the Primary servers to install the software update role.
+        The Primary server computer account must be in the local
+        administrators group to perform the install.
+#>
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName,
+
+        [Parameter()]
+        [Boolean]
+        $AnonymousWsusAccess,
+
+        [Parameter()]
+        [ValidateSet('Internet', 'Intranet', 'InternetAndIntranet')]
+        [String]
+        $ClientConnectionType,
+
+        [Parameter()]
+        [Boolean]
+        $EnableCloudGateway,
+
+        [Parameter()]
+        [Boolean]
+        $UseProxy,
+
+        [Parameter()]
+        [Boolean]
+        $UseProxyForAutoDeploymentRule,
+
+        [Parameter()]
+        [String]
+        $WSUSAccessAccount,
+
+        [Parameter()]
+        [UInt32]
+        $WsusIisPort,
+
+        [Parameter()]
+        [UInt32]
+        $WsusIisSslPort,
+
+        [Parameter()]
+        [Boolean]
+        $WsusSsl,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SiteServerName $SiteServerName
+    $result = $true
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($state.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message ($script:localizedData.SUPNotInstalled -f $SiteServerName)
+            $result = $false
+        }
+        else
+        {
+            $evalList = @('AnonymousWSUSAccess','ClientConnectionType','EnableCloudGateway','UseProxy',
```

Comment:
in the future may want to look at Test-DscParameterState that is in the common helper for future modules.  example of this is in the DistributionPoint Module